packages/actions/test/unit/database.test.ts

```diff
@@ -0,0 +1,288 @@
+import { where } from "firebase/firestore"
+import { CeremonyState } from "@zkmpc/actions/src/types/enums"
+import { fakeCeremoniesData, fakeCircuitsData, fakeParticipantsData, fakeUsersData } from "../data/samples"
+import {
+    getCurrentFirebaseAuthUser,
+    queryCollection,
+    fromQueryToFirebaseDocumentInfo,
+    getAllCollectionDocs,
+    getCurrentContributorContribution,
+    getDocumentById,
+    getCurrentActiveParticipantTimeout,
+    getClosedCeremonies,
+    getParticipantsCollectionPath,
+    getCircuitsCollectionPath,
+    getContributionsCollectionPath,
+    getTimeoutsCollectionPath
+} from "../../src"
+import {
+    setCustomClaims,
+    createNewFirebaseUserWithEmailAndPw,
+    deleteAdminApp,
+    generatePseudoRandomStringOfNumbers,
+    initializeAdminServices,
+    initializeUserServices,
+    sleep
+} from "../utils"
+
+chai.use(chaiAsPromised)
+
+/**
+ * Unit test for Firebase helpers.
+ * @notice some of these methods are used as a core component for authentication.
+ */
+describe("Database", () => {
+    // Init firebase services.
+    const { adminFirestore, adminAuth } = initializeAdminServices()
+    const { userApp, userFirestore } = initializeUserServices()
+    const userAuth = getAuth(userApp)
+
+    // Sample data for running the test.
+    const user = fakeUsersData.fakeUser2
+    const coordinatorEmail = "coordinator@coordinator.com"
+    // storing the uid so we can delete the user after the test
+    let coordinatorUid: string
+
+    // generate passwords for user and coordinator
+    const userPwd = generatePseudoRandomStringOfNumbers(24)
+    const coordinatorPwd = generatePseudoRandomStringOfNumbers(24)
+
+    beforeAll(async () => {
+        // create a new user without contributor privileges
+        await createNewFirebaseUserWithEmailAndPw(userApp, user.data.email, userPwd)
+        await sleep(5000)
+
+        // Retrieve the current auth user in Firebase.
+        const currentAuthenticatedUser = getCurrentFirebaseAuthUser(userApp)
+        user.uid = currentAuthenticatedUser.uid
+
+        // create account for coordinator
+        await createNewFirebaseUserWithEmailAndPw(userApp, coordinatorEmail, coordinatorPwd)
+        await sleep(5000)
+
+        const currentAuthenticatedCoordinator = getCurrentFirebaseAuthUser(userApp)
+        coordinatorUid = currentAuthenticatedCoordinator.uid
+
+        // add custom claims for coordinator privileges
+        await setCustomClaims(adminAuth, coordinatorUid, { coordinator: true })
+
+        // Create the mock data on Firestore.
+        await adminFirestore
+            .collection(`ceremonies`)
+            .doc(fakeCeremoniesData.fakeCeremonyOpenedFixed.uid)
+            .set({
+                ...fakeCeremoniesData.fakeCeremonyOpenedFixed.data
+            })
+
+        await adminFirestore
+            .collection(`ceremonies/${fakeCeremoniesData.fakeCeremonyOpenedFixed.uid}/circuits`)
```

Comment:
Please, be sure to use new methods for path rules.

For e.g., here you should remove the manual path and use instead `getCircuitsCollectionPath(fakeCeremoniesData.fakeCeremonyOpenedFixed.uid)` importing from `actions` package (use src path).